cli/src/command_error.rs

```diff
@@ -295,12 +295,27 @@ impl From<OpsetEvaluationError> for CommandError {
 impl From<SnapshotError> for CommandError {
     fn from(err: SnapshotError) -> Self {
         match err {
-            SnapshotError::NewFileTooLarge { .. } => {
-                user_error_with_message("Failed to snapshot the working copy", err).hinted(
-                    r#"Increase the value of the `snapshot.max-new-file-size` config option if you
-want this file to be snapshotted. Otherwise add it to your `.gitignore` file."#,
-                )
-            }
+            SnapshotError::NewFileTooLarge {
+                path,
+                size,
+                max_size,
+            } => user_error(format!(
+                "Failed to snapshot the working copy\nThe file '{}' is too large to be \
+                 snapshotted ({} bytes).\nThe maximum size allowed is {} bytes.",
+                path.display(),
+                size.0,
+                max_size.0
```

Comment:
iirc, the original intent was to show human-readable size in the error message (because the exact size wouldn't be important.) If we decided to show the exact size, maybe we should also change the error variant to not use `HumanByteSize`.

Reply:
Yes, but that can result in some awkward errors when the difference between them is within epsilon; take the example from the given discussion I linked:

> of size ~1.0MiB exceeds snapshot.max-new-file-size (1.0MiB)

I don't think this is ideal. So maybe we should show the human size if they're more than some delta off, but show the absolute size otherwise? That's probably doable.

Reply:
> Yes, but that can result in some awkward errors when the difference between them is within epsilon; take the example from the given discussion I linked:
> 
> > ... of size ~1.0MiB exceeds snapshot.max-new-file-size (1.0MiB)
> 
> I don't think this is ideal.

I feel the same, and I generally prefer simpler code because I personally don't care much about the readability of these size parameters.

> So maybe we should show the human size if they're more than some delta off, but show the absolute size otherwise? That's doable.

That seems also okay. Since we'll do non-trivial formatting in the CLI layer, I think it's better to remove `HumanByteSize` from the error variant.

Reply:
If we output plain bytes, it'd be nice to add separators, e.g. `1_234_567 bytes`.

Then, it'd also become nice to allow underscores in the input.